### ISSUE TITLE
Allow to specify mode for `git status --untracked-files=normal`

### ISSUE THREAD

**Issue by DmT021** [ForkIssues/Tracker#918]
As far as I can see Fork always use `git status --untracked-files=all`. And it runs the command very often.
Since I'm working with a big enough repos (tens of gigabytes, millions objects) `git status --untracked-files=all` takes a lot of time which makes Fork almost unusable.
It would be nice if I can to specify somehow `mode` to `no|normal|all`, because in most cases `normal` is the mode I want.

**Comment by DanPristupov** [comment]
You can make `--untracked-files=no` in file list preferences. It's called 'Hide untracked files'

<img width="383" alt="Screenshot 2020-02-27 at 14 49 15" src="https://user-images.githubusercontent.com/618115/75451169-55dd5800-5970-11ea-9cdc-ab427f77dbc4.png">

**Comment by DmT021** [comment]
Wow, I didn't know that! Thank you! 
Is it possible to support `normal` mode as well?

**Comment by DanPristupov** [comment]
I didn't know about `normal` before. I just briefly checked and it looks like we can't use it because it will require too many changes in Fork.

However I will measure how fast it is. If it really fast, may be it makes sense to spend time on that.

**Comment by DmT021** [comment]
I can share my measures.
3 times of `time git status -uall`:
```
git status -uall  2.94s user 2.80s system 53% cpu 10.721 total
git status -uall  3.47s user 3.43s system 49% cpu 13.920 total
git status -uall  3.35s user 3.09s system 49% cpu 12.952 total
```
vs 3 times of `time git status -unormal`:
```
git status -unormal  1.81s user 0.38s system 139% cpu 1.566 total
git status -unormal  1.93s user 0.44s system 134% cpu 1.758 total
git status -unormal  1.79s user 0.39s system 117% cpu 1.853 total
```
vs 3 times of `time git status -uno`:
```
git status -uno  0.58s user 0.43s system 149% cpu 0.678 total
git status -uno  0.59s user 0.43s system 149% cpu 0.684 total
git status -uno  0.54s user 0.43s system 132% cpu 0.733 total
```

**Comment by DanPristupov** [comment]
Can you add `time git status -uno` for comparison please?

**Comment by DmT021** [comment]
Sure. Updated the comment.

**Comment by DanPristupov** [comment]
OK, looks like it is almost twice faster than `-uall`. Thank you!

Don't expect any changes in the near future, but I will be thinking about a way to take such optimization into account.

**Comment by DmT021** [comment]
Ok, thank you!

**Comment by DanPristupov** [comment]
Actually... It's really great we have this conversation now. I didn't even know people with such big repos use Fork.

I think I can *significantly* improve general performance for you and may be you will not even need `-unormal`. Hopefully I'll have a time for this in 1.0.91/1.0.92. I'll let you know.

**Comment by saitjr** [comment]
Hi, is there any update for this issue? still missing normal flag in version 2.49.1.

We also work with a huge repository and have enabled the `core.untrackedCache` optimization. However, Fork uses `--untracked-files=all`, which causes Git to ignore this optimization.

here is git source code:

<img width="893" alt="Image" src="https://github.com/user-attachments/assets/c60d38d3-2749-4c3e-99be-9efc47efda13" />

**Comment by DanPristupov** [comment]
@saitjr 

> is there any update for this issue?

No, if I remember correctly, I didn't notice any significant improvement due to this option.

Can you show the measures (https://github.com/fork-dev/Tracker/issues/918#issuecomment-592022603)?

**Comment by saitjr** [comment]
thank for you reply, here is the data:

3 times of time git status -uall:
```
git status -uall  0.46s user 2.98s system 31% cpu 10.887 total
git status -uall  0.53s user 3.36s system 35% cpu 10.839 total
git status -uall  0.48s user 2.97s system 31% cpu 10.784 total
```

vs 3 times of time git status -unormal:
```
git status -unormal  0.29s user 0.13s system 87% cpu 0.473 total
git status -unormal  0.15s user 0.10s system 97% cpu 0.255 total
git status -unormal  0.15s user 0.11s system 80% cpu 0.329 total
```

vs 3 times of time git status -uno:
```
git status -uno  0.05s user 0.08s system 55% cpu 0.247 total
git status -uno  0.06s user 0.08s system 62% cpu 0.221 total
git status -uno  0.06s user 0.07s system 70% cpu 0.186 total
```

for total time, use `--untracked-files=all` will takes 11s, and use `git status` without argv takes 0.3s

**Comment by saitjr** [comment]
Using `--untracked-files=normal` will hide the folder structure if the folder is not tracked.

<img width="1102" alt="Image" src="https://github.com/user-attachments/assets/eb26bcda-37b1-4dcb-a342-18ce29111196" />

If you use `--untracked-files=normal` instead of `--untracked-files=all`, you will need to manually expand subfolders. If there are untracked folders, use `all`; otherwise, use `normal`.

<img width="616" alt="Image" src="https://github.com/user-attachments/assets/8c0a32d2-5c6d-4e3a-b9df-4e07c5ef0f7f" />

**Comment by saitjr** [comment]
You can use `git ls-files` to manually list untracked files, as shown in the image below:

1. Run `git status` to identify untracked folders.
2. Use `git ls-files --others --exclude-standard <folder>` to list all untracked files within the folder.

This method is faster than using git status --untracked-files=all.

<img width="1268" alt="Image" src="https://github.com/user-attachments/assets/333eee23-c70b-44da-abc4-27bfa2d61d27" />

**Comment by saitjr** [comment]
Is there any possibility for this feature? 🥹
